app/src/main/java/agency/five/codebase/android/movieapp/ui/favorites/FavoritesViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package agency.five.codebase.android.movieapp.ui.favorites
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.ui.favorites.mapper.FavoritesMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class FavoritesViewModel(
+    private val movieRepository: MovieRepository,
+    favoritesMapper: FavoritesMapper,
+) : ViewModel() {
+    val favoritesViewState: StateFlow<FavoritesViewState> = movieRepository.favoriteMovies()
+        .map { movies -> favoritesMapper.toFavoritesViewState(movies) }
+        .stateIn(viewModelScope, SharingStarted.Eagerly, FavoritesViewState(emptyList()))
```

Comment:
`SharingStarted.Eagerly` isn't used very often and has very specific use cases. With it you will listen for updates even when there aren't any collectors (observers). We don't want this in this scenario. `SharingStarted.WhileSubscribed`  would be a better option. `SharingStarted.Lazily` is similar but is a little better because sharing will only start once the first collector comes. 
The application will work fine but this is just a micro-optimization which is very simple too implement so we usually just make it.  
You can read more about it here : 
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-sharing-started/ , 
https://medium.com/androiddevelopers/things-to-know-about-flows-sharein-and-statein-operators-20e6ccb2bc74#:~:text=To%20keep%20the%20underlying%20flow,if%20there%20are%20no%20collectors.